examples/mnist_vae.py

```diff
@@ -0,0 +1,69 @@
+'''Train a simple deep Variational Autoencoder on the MNIST dataset.
```

Comment:
I would recommend adding more detailed commentary on what a Variational Autoencoder is, how it works, and what it can achieve, like a layman version of the paper's abstract.